Sources/Rendering/OpenGL/CellArrayBufferObject/index.js

```diff
@@ -200,13 +231,70 @@ function vtkOpenGLCellArrayBufferObject(publicAPI, model) {
     let vboidx = 0;
     let ucidx = 0;
 
+    // Find out if shift scale should be used
+    // Compute squares of diagonal size and distance from the origin
+    let diagSq = 0.0;
+    let distSq = 0.0;
+    for (let i = 0; i < 3; ++i) {
+      const range = options.points.getRange(i);
+
+      const delta = range[1] - range[0];
+      diagSq += delta * delta;
+
+      const distShift = 0.5 * (range[1] + range[0]);
+      distSq += distShift * distShift;
+    }
+
+    const useShiftAndScale =
+      diagSq > 0 &&
+      (Math.abs(distSq) / diagSq > 1.0e6 || // If data is far from the origin relative to its size
+      Math.abs(Math.log10(diagSq)) > 3.0 || // If the size is huge when not far from the origin
+        (diagSq === 0 && distSq > 1.0e6)); // If data is a point, but far from the origin
+
+    if (!useShiftAndScale && model.coordShiftAndScaleEnabled === true) {
+      // Make sure to reset
+      model.coordShiftAndScaleEnabled = false;
+      model.coordScale = null;
+      model.coordShift = null;
+      model.inverseShiftAndScaleMatrix = null;
```

Comment:
Instead I would call `setCoordShiftAndScale(null, null)` that would set all variables automatically